libvips/foreign/nsgifload.c

```diff
@@ -444,7 +399,9 @@ static int
 vips_foreign_load_nsgif_generate( VipsRegion *or,
 	void *seq, void *a, void *b, gboolean *stop )
 {
-        VipsRect *r = &or->valid;
+	int prev_page = 0;
```

Comment:
If you get prev_page = -1 here you could remove the `bitmap == NULL` from the if() around frame decode below.

Reply:
Done!